conjur/init_controller.py

```diff
@@ -0,0 +1,107 @@
+        self.init_logic = init_logic
+        self.force_overwrite = force
+
+    def call(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        fetched_certificate = self.get_server_certificate(self.init_logic,
+                                                          self.init_input_obj)
+        self.get_host_info(self.init_input_obj)
+        self.write_certificate(self.init_logic,
+                               self.init_input_obj,
+                               self.init_input_obj.cert_file,
+                               fetched_certificate)
+        self.init_logic.write_conjurrc(self.DEFAULT_CONFIG_FILE,
+                                       self.init_input_obj,
+                                       self.force_overwrite)
+
+    # pylint: disable=line-too-long
+    @staticmethod
+    def get_server_certificate(init_logic, input_obj):
+        """
+        Method to get the certificate from the Conjur endpoint detailed by the user
+        """
+        url = input_obj.appliance_url if input_obj.appliance_url else input("Enter the URL of your Conjur service: ")
+        input_obj.appliance_url = ''.join(url)
+
+        url = urlparse(input_obj.appliance_url)
+
+        # Fetch certificate information if connection is expected over TLS
+        certificate_path = input_obj.cert_file if input_obj.cert_file else None
+        if certificate_path is not None:
+            input_obj.cert_file = ''.join(certificate_path)
+
+        if certificate_path is None and url.scheme == "https":
+            logging.info("Initiating a TLS Connection")
+            sock = init_logic.connect_to_socket(url.hostname, url.port)
+            fingerprint, fetched_certificate = init_logic.get_certificate(url.hostname,
+                                                                          url.port, sock,
+                                                                          OpenSSL.crypto.dump_certificate)
+            sys.stdout.write(f"\nSHA1 Fingerprint={fingerprint}\n")
+
+            sys.stdout.write("\nPlease verify this certificate on the appliance using command:\n" \
+                             "\topenssl x509 -fingerprint -noout -in ~conjur/etc/ssl/conjur.pem\n\n")
+            trust_certificate = input("Trust this certificate (yes/no) (default yes): ")
+            if trust_certificate != '' and trust_certificate.lower() != 'yes':
+                raise ValueError("You decided not to trust the certificate")
+
+            return fetched_certificate
+        return None
+
+    @staticmethod
+    def get_host_info(init_input_obj):
+        """
+        Method to host data from the user
+        """
+        host_account = init_input_obj.account if init_input_obj.account else input("Enter your organization account name: ")
+        if not host_account:
+            raise RuntimeError("Error: account is required")
+
+        init_input_obj.account = ''.join(host_account)
```

Comment:
Why is the join needed?

Reply:
casts to string but I will use `str()` here